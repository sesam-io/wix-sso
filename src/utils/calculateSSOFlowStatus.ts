import { SSOFlowStatus } from "../enums";

export type SSOFlowStatusArgs = {
  isAuthenticated: boolean;
  query: string;
};

export const calculateSSOFlowStatus = (args: SSOFlowStatusArgs) => {
  const { isAuthenticated, query } = args;

  if (isAuthenticated) {
    return SSOFlowStatus.LoggedIn;
  }

  if (query.includes("code=") && query.includes("state=")) {
    return SSOFlowStatus.Authenticated;
  }

  return SSOFlowStatus.LoggedOut;
};
