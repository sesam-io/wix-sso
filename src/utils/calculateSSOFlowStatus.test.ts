import { calculateSSOFlowStatus } from "./calculateSSOFlowStatus";

describe("calculateSSOFlowStatus test suite", () => {
  it("should return Authenticated", () => {
    const args = {
      isAuthenticated: false,
      query: "https://www.wave.sesam.io/?code=0yV2lT&state=VFlmX1c",
    };

    expect(calculateSSOFlowStatus(args)).toMatchSnapshot();
  });
});
