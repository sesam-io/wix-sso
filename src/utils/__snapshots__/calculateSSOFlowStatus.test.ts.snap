// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateSSOFlowStatus test suite should return Authenticated 1`] = `"authenticated"`;
